tools/transform.cpp

```diff
@@ -285,6 +285,73 @@ static expr preprocess(Transform &t, const set<expr> &qvars0,
 }
 
 
+static expr
+encode_undef_refinement_per_elem(
+    const Type &ty, const expr &a, const expr &a2, const expr &a_np,
```

Comment:
it seems easier if you pass a/a_np as a stateValue. It avoid 2x evaluations of extract below

Reply:
Fixed